### PR TITLE
Official Drupal 8.9 Upgrade PR

### LINKED ISSUE #2607: Update Drupal Core to next minor version: 8.9.x
## Description 
Drupal 8.7 is EOL and there is a security patch that requires an update to the next minor version. Drupal 8.9 is the LTS version now that 9.0 is out, so we should skip 8.8 and try to get to 8.9 which should be good until 11/2021.

### LINKED ISSUE #2898: Upgrade BLT to 11.x
# Upgrade BLT to 11.x
A BLT 10.x dependency on `doctrine/inflector:~1.1.0` is conflicting with a drupal/recommended version of doctrine/inflector. So BLT needs to be updated to 11.x.  It is unclear if 11.x supports 8.7 or not. I hesitate on attempting both a Drupal Upgrade and a BLT upgrade at the same time. (you know, given our mess around ACSF)

> **ESTIMATE** 20

### LINKED ISSUE #2909: PHPUnit Tests Broke When BLT 10.x was Updated
### Issue description
BLT 10.x changed the meaning of the `path` parameter from being the directory that has tests to being the directory to RUN the tests from. The `path` parameter needs to be changed to be `directory` based on the [blt testing docs](https://docs.acquia.com/blt/developer/testing/#phpunit).

> **ESTIMATE** NN

### PR DESCRIPTION
**Pull request by bryanpizzillo**
## (#2898,#2607,#2909) Drupal 8.9 Update
- Updated Drupal to 8.9.6
   - Rerolled workflow patches into an "unofficial" patch
   - Resolved a number of deprecated code notices
   - Fixed a bad hook update in cgov_caching_nocdn because the name was actually being validated after update
   - Updated a number of features packages after core changes resulted in new configuration files
   - Hid new status field on taxonomy terms.  This is a new field added now that taxonomy_terms are revisable entities. We are hiding it, and the default value is Published for each revision. IF a term is maked as unpublished, the systems will not behave as expected. For example, Site Sections will have their navigation label updated on the live site even when saving with an unpublished status. The same site section though cannot be found in the site section lookup for a node after being unpublished.
   - Updated translations
- Updated BLT to 11.x
   - Resolved issue with BLT namespaces changes caused issue with CgovConfigCommands
   - Use OOB pipelines build_artifact script as we were no longer in sync with the one from BLT 11
- Switched from webflo/drupal-core-require-dev to drupal/core-dev for development dependencies
- Switched from drupal-composer/drupal-scaffold to drupal/core-composer-scaffold for copying in core assets on composer install of Drupal (e.g. .htaccess)
- Updated MANY modules - see [Third Party Module Changes D8.7->D8.9.pdf](https://github.com/NCIOCPL/cgov-digital-platform/files/5271544/Third.Party.Module.Changes.D8.7-.D8.9.pdf) for details. Implemented the following fixes due to changes:
   - Needed to change how we move buttons around in moderation_sidebar
   - Added ThemeNegotiator to around a change in entity_embed introduced in 2844822. NOTE: This will need to be fixed up when 3108093 gets fixed as it changes the route name for the preview controller. There are also a number of issues relating to how there should be some way to override the embed template when previewing. However, none of those issues have a resolution, or even agreement on an approach.
   - embed module icons are now base64 encoded in stored in the config. Old icons have been kept in order to have easy access when we need to update these configs, but those icons are no longer used by the module.
   - Fixed config change introduced after address 1.4
- Fixed issue with PHPUnit in Pipelines trying to write to the root directory and causing errors.
- Removed a number of patches as the updates incorporated those changes.
- Fix PHPUnit config where `blt test:phpunit` failed

>**_NOTE:_** In order to allow the code to deploy to an existing environment after the changes to core's path_alias module we had to turn off the new drupal-check deprecated code checks in BLT 11. This should be resolved after this release with #2918.

Closes #2898
Closes #2607
Closes #2909

## (#2917) Fix cache info being cleared from entities
This issue was uncovered by the entity_browser module update, however this commit could stand alone, so I left it separate.
- The fix added in new complications as items started to be cached correctly, but pre-existing issues resulted in incorrect cache tags ending up in render_cache.
- Added Twig Extension to Force Cache Bubbling When content variable is not directly used in twig templates.
- Fixed issue with list items templates where cache tags were not bubbling, and therefore changes to node titles would trigger a cache clear of the pages with the lists, but did not clear out the rendered list item itself.

Closes #2898

### CODE REVIEW

**Review by blairlearn — CHANGES_REQUESTED** [review]

**Comment by VictoriaSunNIH** [comment]
The Tasks bar has changed. View in Edit form is no longer showing above "Show Revisions".  And "Edit content" appears now above workflow actions.  This edit content should be moved to be below workflow actions, and above show revisions, as well as renamed "view in edit form."
image

**DEV (this is correct):**
image


**TEST (this is incorrect):**

**Comment by bryanpizzillo** [comment]
Here is a list of the various third party module changes for this release.
[Third Party Module Changes D8.7->D8.9.pdf](https://github.com/NCIOCPL/cgov-digital-platform/files/5271544/Third.Party.Module.Changes.D8.7-.D8.9.pdf)

File: docroot/.htaccess.orig [hunk 1]
```diff
@@ -0,0 +1,198 @@
+#
```
[the remark this answers is not in the document]
Comment (bryanpizzillo) [reply docroot/.htaccess.orig]: I removed it and added to git ignore

**Comment by bryanpizzillo** [comment]
> The Tasks bar has changed. View in Edit form is no longer showing above "Show Revisions". And "Edit content" appears now above workflow actions. This edit content should be moved to be below workflow actions, and above show revisions, as well as renamed "view in edit form."
> 

@VictoriaSunNIH  can you please test this on http://ncigovcdode385.prod.acquia-sites.com/? I updated everything and I think it is right now.

**Comment by blairlearn** [comment]
An "oddity" -- I'm seeing that some -- but not all -- users on the test environment no longer have their email addresses set.

e.g. user `olssonja` has no email address.  Ditto for `tanenbaumag`.  On the other hand, `welshja` and `mashkinaea` have the sort of "filler" addresses one would expect from staging to the  lower tier.

**Comment by VictoriaSunNIH** [comment]
Promotional Image and Override Promotional Image - no matter what image you select, the Drupal field says 'Breast Cancer in Women'. The front-end will display the correct image, but the Promotional Image field does not display the correct image name here.

**Comment by VictoriaSunNIH** [comment]
@bryanpizzillo  - another UI change that we need to address: For all the content types, under URL Redirects on the right hand side of the CMS interfact, there is a new Extra button called "Add URL Redirects" appearing.  This change needs to be addressed.  We should hide that button since only IAs are allowed to do this
image

**Comment by lburack** [comment]
@bryanpizzillo - the video carousel controls are missing at mobile breakpoints on the front end. On mobile, it displays as though there's only one video; there's no way to see that this is a carousel, including inability to swipe. 

https://www-test-acsf.cancer.gov/news-events/press-releases/2015/mini-test
versus
https://www.cancer.gov/types/childhood-cancers (video is under last OTP section, Research)

video-carousel-mobile

**Pull request merged**